core/ArrayMacros.carp

```diff
@@ -0,0 +1,44 @@
+(defmacro doall [f xs]
```

Comment:
Could this just go into `Array.carp`, maybe?

Reply:
yes!

Reply:
Or oh, you mean the whole file? Yeah I tried that at first but there were macros that were used by the stack allocated Array that made it not work. I'll try again though.